packages/openmpi/package.py

```diff
@@ -0,0 +1,699 @@
+    #   1) https://github.com/open-mpi/ompi/pull/2294
+    #   2) https://github.com/open-mpi/ompi/pull/5099
+    # The first one was applied starting version v3.0.0 and backported to
+    # v1.10. A subset with relevant modifications is applicable starting
+    # version 1.8.4.
+    patch('use_mpi_tkr_sizeof/step_1.patch', when='@1.8.4:1.10.6,2:2.999')
+    # The second patch was applied starting version v4.0.0 and backported to
+    # v2.x, v3.0.x, and v3.1.x.
+    patch('use_mpi_tkr_sizeof/step_2.patch', when='@1.8.4:2.1.3,3:3.0.1')
+
+    variant(
+        'fabrics',
+        values=disjoint_sets(
+            ('auto',),
+            ('psm', 'psm2', 'verbs',
+             'mxm', 'ucx', 'ofi',
+             'fca', 'hcoll',
+             'xpmem', 'cma', 'knem')  # shared memory transports
+        ).with_non_feature_values('auto', 'none'),
+        description="List of fabrics that are enabled; "
+                    "'auto' lets openmpi determine",
+    )
+
+    variant(
+        'schedulers',
+        values=disjoint_sets(
+            ('auto',),
+            ('alps', 'lsf', 'tm',
+             'slurm', 'sge', 'loadleveler')
+        ).with_non_feature_values('auto', 'none'),
+        description="List of schedulers for which support is enabled; "
+                    "'auto' lets openmpi determine",
+    )
+
+    # Additional support options
+    variant('atomics', default=False, description='Enable built-in atomics')
+    variant('java', default=False, description='Build Java support')
+    variant('static', default=True, description='Build static libraries')
+    variant('sqlite3', default=False, description='Build SQLite3 support')
+    variant('vt', default=True, description='Build VampirTrace support')
+    variant('thread_multiple', default=False,
+            description='Enable MPI_THREAD_MULTIPLE support')
+    variant('cuda', default=False, description='Enable CUDA support')
+    variant('pmi', default=False, description='Enable PMI support')
+    variant('wrapper-rpath', default=True,
+            description='Enable rpath support in the wrappers')
+    variant('cxx', default=False, description='Enable C++ MPI bindings')
+    variant('cxx_exceptions', default=False, description='Enable C++ Exception support')
+    variant('gpfs', default=True, description='Enable GPFS support (if present)')
+    # Adding support to build a debug version of OpenMPI that activates
+    # Memchecker, as described here:
+    #
+    # https://www.open-mpi.org/faq/?category=debugging#memchecker_what
+    #
+    # This option degrades run-time support, and thus is disabled by default
+    variant(
+        'memchecker',
+        default=False,
+        description='Memchecker support for debugging [degrades performance]'
+    )
+
+    variant(
+        'legacylaunchers',
+        default=False,
+        description='Do not remove mpirun/mpiexec when building with slurm'
+    )
+    
+    provides('mpicuda', when='+cuda')
```

Comment:
Very interesting. Of course it's unfortunate that we have have to _fork_ the openmpi package due to the maintenance overhead. But since this is only one of various _sub-optimal_ solutions, it's fine. Maybe we'll figure out a clean solution eventually.